tensorflow/contrib/repeat/kernels/repeat_op_cpu_impl.cc

```diff
@@ -0,0 +1,236 @@
+/* Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/util/work_sharder.h"
+
+namespace tensorflow{
+
+namespace {
+
+template <typename T>
+struct MemCopier {
+  
+  inline void Copy(const T* input, T* output, int64 size){
```

Comment:
I'm not sure reparameterizing copy improves readability or not? If you have a good argument, keep it, but explain it in comment here.